cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -88,6 +90,14 @@ func newASGCache(service autoScalingWrapper, explicitSpecs []string, autoDiscove
 	return registry, nil
 }
 
+var getInstanceTypeForAsg = func(m *asgCache, asg *asg) (string, error) {
+	if obj, found, _ := m.asgInstanceTypeCache.GetByKey(asg.AwsRef.Name); found {
+		return obj.(instanceTypeCachedObject).instanceType, nil
+	}
+
+	return "", fmt.Errorf("Could not find instance type for %s", asg.AwsRef.Name)
```

Comment:
When a new launch template (name, but new version) is released, could this error out because populate function has not run to fill cache?

Should we fallback here to fill cache?

Reply:
I don't believe this is necessary; the `RunOnce` function [calls](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/static_autoscaler.go#L255) `CloudProvider.Refresh()`, which in turn re-populates the cache.  `RunOnce` is called on every iteration, so the only time the situation you describe can happen is if the ASG is created between when `Refresh` is called and when `ScaleUp` [calls](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/scale_up.go#L499) `GetNodeInfoFromTemplate` (which eventually filters down into this function).  The error here is caught and handled, so the autoscaler will just skip the ASG on this iteration.  On the next iteration, `Refresh()` will be called again and it will pick up the new LT.